### PR TITLE
RTL8822CU-aml: new package

### PR DESCRIPTION
**Pull request by kszaq**
Adds RTL8822CU 802.11ac USB WiFi driver. Compile and runtime tested with Amlogic-ng target and a cheap USB WiFi stick. Not tested with Amlogic-ne (I have no targets).

Purpose of having this driver: another way to add 802.11ac radio to boxes that don't have it.

### CODE REVIEW

**Comment by Portisch** [comment]
As you picked branch `khadas-vim4-r-64bit` it should also compile for 5.4, `Amlogic-ne` + 5.15 `Amlogic-n?`. Let me check it tomorrow... Thx

**Comment by kszaq** [comment]
Yes, it should work but I have no targets to test it with.

On the other hand, for kernels >=5.4, there is a much more up to date driver available: https://github.com/lwfinger/rtw88

Also regarding Realtek, I think you can drop `RTL8812AU` and `RTL8814AU` from `projects/Amlogic-ce/packages/linux-drivers` (both no longer maintained) and replace them with `packages/linux-drivers/RTL8812AU` - I checked supported hardware IDs and the latter covers all IDs from the unmaintained packages + additional 2 IDs. I can PR that change, if you want?

File: projects/Amlogic-ce/packages/linux-drivers/amlogic/RTL8822CU-aml/package.mk [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# SPDX-License-Identifier: GPL-2.0
+# Copyright (C) 2017-present Team LibreELEC (https://libreelec.tv)
+
+PKG_NAME="RTL8822CU-aml"
+PKG_VERSION="f1a4a3abb993245bf28bf1e2254f47f76972fd6d"
+PKG_SHA256="2ad96072107478964b9308c19fa0edefdbb70e4c088256432b6fe5b903ebcf35"
+PKG_ARCH="arm aarch64"
+PKG_LICENSE="GPL"
+PKG_SITE="https://github.com/khadas/android_vendor_wifi_driver_realtek_8822cu/"
+PKG_URL="https://github.com/khadas/android_vendor_wifi_driver_realtek_8822cu//archive/$PKG_VERSION.tar.gz"
```
Comment (Portisch) [inline projects/Amlogic-ce/packages/linux-drivers/amlogic/RTL8822CU-aml/package.mk]: Please use current used formatting like `${PKG_VERSION}` as example.
Also there are two `//` in the `PKG_URL`.

**Comment by Portisch** [comment]
Built with 5.15:
```text
alias usb:v13B1p0043d*dc*dsc*dp*icFFiscFFipFFin* 88x2cu
alias usb:v0BDApD82Bd*dc*dsc*dp*icFFiscFFipFFin* 88x2cu
alias usb:v0BDApD820d*dc*dsc*dp*icFFiscFFipFFin* 88x2cu
alias usb:v0BDApC812d*dc*dsc*dp*icFFiscFFipFFin* 88x2cu
alias usb:v0BDApC82Ed*dc*dsc*dp*icFFiscFFipFFin* 88x2cu
alias usb:v0BDApC82Cd*dc*dsc*dp*icFFiscFFipFFin* 88x2cu
```

Please add distribution dependency:
```diff
diff --git a/distributions/CoreELEC/options b/distributions/CoreELEC/options
index 59dd3bb0a0..e7d44537a2 100644
--- a/distributions/CoreELEC/options
+++ b/distributions/CoreELEC/options
@@ -58,7 +58,7 @@
 # Space separated list is supported,
 # e.g. ADDITIONAL_DRIVERS="DRIVER1 DRIVER2"
   ADDITIONAL_DRIVERS="gpu-aml openvfd-driver media_modules-aml wifi_dummy-aml ap6xxx-aml
-                     uwe5631-aml RTL8822CS-aml RTL8852BS-aml RTL8152-aml w1-aml"
+                     uwe5631-aml RTL8822CS-aml RTL8822CU-aml RTL8852BS-aml RTL8152-aml w1-aml"
 
 # Default size of system partition, in MB, eg. 512
   SYSTEM_SIZE=512
```
Then the driver should be included in Amlogic-ng/-ng-dv/ne/no

**Comment by Portisch** [comment]
> I can PR that change, if you want?

Please do so, thx!

**Comment by kszaq** [comment]
Thank you for the review. Done. 🙂

**Comment by Portisch** [comment]
I meant to enclose all variables in `${...}` :wink:
Then it should be ready for merge, thx!

**Comment by kszaq** [comment]
Sorry, it was a simple c/p from [RTL8822BU-aml](https://github.com/CoreELEC/CoreELEC/blob/coreelec-21/projects/Amlogic-ce/packages/linux-drivers/amlogic/RTL8822BU-aml/package.mk). I will fix it today evening!

**Comment by kszaq** [comment]
Done.

**Pull request merged**